packages/core/src/services/scales-cartesian.ts

```diff
@@ -239,26 +303,24 @@ export class CartesianScales extends Service {
 		return this.getScaleTypeByPosition(this.getMainYAxisPosition());
 	}
 
-	getDomainIdentifier() {
+	getDomainIdentifier(data?: any) {
 		const options = this.model.getOptions();
-		const axisOptions = Tools.getProperty(
+		return Tools.getProperty(
 			options,
 			"axes",
-			this.domainAxisPosition
+			this.getDomainAxisPosition(data),
+			"mapsTo"
 		);
-
-		return axisOptions.mapsTo;
 	}
 
-	getRangeIdentifier() {
+	getRangeIdentifier(data?: any) {
 		const options = this.model.getOptions();
-		const axisOptions = Tools.getProperty(
+		return Tools.getProperty(
 			options,
 			"axes",
-			this.rangeAxisPosition
+			this.getRangeAxisPosition({datum: data}),
```

Comment:
is `data` called `data` or `datum`? Seems like it should be the same in both places. If it's a datapoint then `datum`, and if it's multiple then `data`

Reply:
good call